woocommerce/woocommerce
# Add product link field

### All Submissions:






### Changes proposed in this Pull Request:

This PR add's a new product link field under the name field.
Some remaining work will have to be done to add the `permalink_template` to the product REST api so we don't need retrieve this separately - here is the follow up issue - https://github.com/woocommerce/woocommerce/issues/34312. 

Addresses: https://github.com/woocommerce/woocommerce/issues/34029

https://user-images.githubusercontent.com/2240960/184427098-11d5ab38-c4ba-4a58-b397-fc6e48e2f652.mp4

### How to test the changes in this Pull Request:

You can double check the designs with this link: 5sAIeTRd9Yp7nSCT33BAWz-fi-5175%3A104437

1. Load this branch and enable the `new-product-management-experience` feature flag by using the WCA Test Helper (Tools > WCA Test Helper > Features) and refresh the page.
2. Click the **Products > Add New (MVP)** button and fill in the name field and hit **Save draft**
3. This should show the product link text underneath the name field with a button that say's `Edit`
4. Click `edit`, this should popup a modal allowing you to edit the slug.
5.  Update the slug field and hit **Save**, it should save and close the modal, it will show a notice emphasizing that the product link has been successfully saved. It will also output a `wcadmin_product_update_slug` track, if you have your console open with `localStorage.setItem( 'debug', 'wc-admin:*' );` enabled.
6. The field should have updated with the new product slug.
7. Click `edit` again and make sure the close and cancel buttons work as expected.
8. Now edit the product name field and click the `edit` button again and update the slug.
9. Once the modal closes the product link should of updated, and the name field should still contain your previous changes with the **Update** button enabled to save these.
10. You can also **publish** the product and follow the same steps.

### Other information:




### FOR PR REVIEWER ONLY:
